### DISCUSSION TITLE
third-party interpretation tool integration

### DISCUSSION THREAD

**Discussion by parlar** [Clinical-Genomics/scout#4804] [Ideas]
In Umeå, Scout has been selected as the primary tool for genetic variant interpretation and as a repository/database for variant information. However, there has been a request to also have the option to also use additional third-party interpretation tools, such as Qiagen QCI, Genoox Franklin, Roche Navify Mutation Profiler. This may prove useful in tricky cases. To facilitate this, we need a way to transfer data from Scout to such tools.

Scout provides a comprehensive environment for variant interpretation. It is our understanding that the Scout development team prefers to keep the core system streamlined, suggesting that additional functionality should be implemented as “extensions” rather than directly into Scout itself. 

The planned feature is quite simple. It should enable selection of all, or a subset of genetic variants linked to a case, add additional metadata required by the third-party tool, and facilitate the transfer of the data and metadata to the third-party tool. Maybe in the future functionality for retrieval of results from third-party tools could be added but that will come later.

The problem is only how to do it. We have no previous experience of developing Scout features and extensions and therefore do not fully understand of the limitations of the Scout extension system. We are however very happy that we have been able to hire a consultant, Fredrik Silvearv, to work on this. (Fredrik has a background in quantum mechanics/materials science and is a good Python programmer.)   

I was hoping that we might get some suggestions for a good way to implement this from the Scout dev team. We had a look at chanjo, which seems to be quite integrated into Scout itself despite being an extension. Would chanjo be a good example to follow or is that not how it would be done today?

Would it somehow be possible to put views into an extension, to avoid modifying the Scout server code too much. Or is this not possible? Or will it have to be done as a separate application, integrated via an extension. It's a small feature, would like to keep it a simple as possible.

**Comment by dnil** [comment]
Hi Pär! Good to hear, and extra hands and brains are always much needed! We welcome plugins for external tools; chanjo is a reasonable template, but perhaps matchmaker or phenopacket will be more what you are looking for. We will certainly be available for for discussions and review, but I just want to repeat a couple of points to make sure you don't waste your effort.
- If the external tool is to do variant analysis for a case, it makes little sense to first run your own pipe, then import a few limited vars to scout select and export/api query to your external tool and get results back. The ideal workflow would be the external tool is fed raw data, does selection without loosing variants as potential false negatives to prior pipe and exports variants of interest for manual inspection in scout.
- Not taking notice of the tools in question, there has several times been discussion of tools that are not necessarily legally allowed for a basic clinical case, only specifically and carefully consented research cases. For such, we should limit visibility to cases where it is allowed.

**Comment by dnil** [comment]
(And e.g. that mutation profile idea from the discussion in https://github.com/Clinical-Genomics/scout/issues/4515 is still up for grabs! 😉 Also, there is that nice new telomere counting tool I've been hearing about from Umeå - loading results from it would be really cool!)

**Comment by parlar** [comment]
Yes! Should perhaps have referenced that discussion. Thanks also for answering quickly. 

I should state that I'm actually not working in Scout and I may lack in understanding of it. (Niklas Mähler would know more.) The request for more options when it comes to interpretation tools comes from the MD:s and clinical scientists. Trying to comply.

It might be that we should not upload the same data that is viewed in Scout, but rather full datasets. But I guess that this could still be managed through a Scout extension but then perhaps preferably as a separate application linked to Scout . And as I understand QCI Interpret, it is still panel-based and the cost depends on the size of the panel. If I have not misunderstood Qiagen, that is. 

But Qiagen and the other companies also have their own proprietary databases and include additional things like drugs and trials for cancer (which as I believe is not provided in Scout right now. I should perhaps work on that ...) Then it's like a second opinion. And for tricky cases it might also save time, when a lot of digging is required.

Regarding the tools in question, it is clinical CE-IVD tools that we are considering.

But we are committed to Scout and would like to keep things together around Scout as much as possible, even if other tools are used also.

**Comment by dnil** [comment]
Again, just go ahead if you need it and have thought the use cases through. 👷 I see clear use for sending full genomes, typically by automation, and quickly retrieving a few highly likely candidates for evaluation (typically easy cases) via some api of theirs to Scout. Would be lovely, and likely a very good case for an extension! Annotating somatic variants with druggability etc is also likely a good commercial use-case; the public dbs on that side appear a bit old and noisy so far, though there are efforts to change it. Ideally as something we would call on during pipeline annotation naturally, but if that is just not possible... 🤷‍♂️  Caution with the US cloud ones, but EU CE-IVD sound very kosher indeed. 😸

**Comment by vwirta** [comment]
> But Qiagen and the other companies also have their own proprietary databases and include additional things like drugs and trials for cancer (which as I believe is not provided in Scout right now. I should perhaps work on that ...) Then it's like a second opinion. And for tricky cases it might also save time, when a lot of digging is required.

Quick comment regarding drugs and trials... for this purpose David Tamberero, Janne Lehtiö and colleagues at KI/Scilifelab have developed the MTB Portal. Might be worthwhile to look into that as well
https://www.mtbp.org/

**Comment by dnil** [comment]
Mm, they didn’t open source it, published it, but are not sharing it on reasonable request as they claim. Something is not quite right there.

**Comment by dnil** [comment]
It could just be me/us naturally - it can be a bit easier to collaborate at a distance! 😊

**Comment by dnil** [comment]
I've transferred this to a discussion - feel very free to add specific third party components/tools as issues and PRs, and we can do them one by one in more detail!
